backend/de.metas.dunning/src/main/java/de/metas/dunning/api/impl/DunningBL.java

```diff
@@ -247,6 +252,63 @@ public void processCandidates(final IDunningContext context, final IDunningCandi
 		}
 
 		dunningProducer.finish();
+
+		sendMassNotifications(dunningProducer.getCreatedDunningDocIds(), Language.getBaseLanguage());
+	}
+
+	private void sendMassNotifications(@NonNull final Collection<DunningDocId> createdDunningDocIds,
+			@NonNull final Language language)
+	{
+		dunningDAO.getByIdsInTrx(createdDunningDocIds)
+				.stream()
+				.collect(Collectors.groupingBy(doc -> docTypeDAO.getById(DocTypeId.ofRepoId(doc.getC_DocType_ID())).getMassGenerateNotification()))
+				.forEach((boilerPlateWithLineId, dunningDocs)-> sendMassNotifications(boilerPlateWithLineId,dunningDocs, language));
+	}
+
+	private void sendMassNotifications(@NonNull final BoilerPlateWithLineId boilerPlateWithLineId,
+			@NonNull final List<I_C_DunningDoc> dunningDocs,
+			@NonNull final Language language)
+	{
+		dunningDocs.stream()
+				.collect(Collectors.groupingBy(doc -> OrgId.ofRepoId(doc.getAD_Org_ID())))
+				.forEach((orgId, docs) -> this.sendMassNotifications(boilerPlateWithLineId, orgId, docs, language));
+	}
+
+	private void sendMassNotifications(final @NonNull BoilerPlateWithLineId boilerPlateWithLineId,
+			@NonNull final OrgId orgId,
+			@NonNull final List<I_C_DunningDoc> dunningDocs,
+			@NonNull final Language language)
+	{
+		final BoilerPlate headerBoilerPlate = boilerPlateRepository.getByBoilerPlateId(boilerPlateWithLineId.getHeaderId(), language);
+		final String lines = getMassGenerateLinesPlainString(boilerPlateWithLineId, dunningDocs, language);
+
+		final Evaluatee headerEvaluatee = Evaluatees.ofSingleton(MASS_GENERATE_LINES_EVALUATEE_PARAM, lines)
+				.andComposeWith(getPO(orgDAO.getById(orgId)));
```

Comment:
what's the reason of composing with AD_Org PO?
it feels a bit awkward... maybe you want to just extract some infos from it.

Reply:
it's for the @Name@ variable used in both subject and content.. I did not want to add extra hardcoded variables.

Reply:
so, when defining the line template, the user will use the `@Name@` as a placeholder for organization name?

Reply:
TBH, i don't like it, but leave it like that for now